analyzers/src/SonarAnalyzer.CSharp/Rules/ParameterValidationInMethodWalker.cs

```diff
@@ -61,16 +61,22 @@ public override void VisitThrowStatement(ThrowStatementSyntax node)
             {
                 argumentExceptionLocations.Add(node.Expression.GetLocation());
             }
+
             // there is no need to visit children
         }
 
         public override void VisitInvocationExpression(InvocationExpressionSyntax node)
         {
             if (node.IsMemberAccessOnKnownType("ThrowIfNull", KnownType.System_ArgumentNullException, semanticModel))
             {
+                // "ThrowIfNull" returns void so it cannot be an argument. We can stop.
                 argumentExceptionLocations.Add(node.GetLocation());
             }
-            // "ThrowIfNull" returns void so it cannot be an argument. We can stop.
+            else
```

Comment:
Without this `else` block, the visit recursion stops for this node. 
This means that for something like:
```csharp
whatever.Invocation()
```
The `whatever.Invocation` syntax node is not going to keep visiting inside the `whatever` node, in which case if `whatever` contains an `AwaitExpression`, we will not spot it to stop the SyntaxWalker via:
```csharp
public override void VisitAwaitExpression(AwaitExpressionSyntax node) =>keepWalking = false;
```
Which produces this the FP in the testcases.